eng/common/TestResources/New-TestResources.ps1

```diff
@@ -259,6 +262,9 @@ if ($TestApplicationSecret) {
 if ($AdditionalParameters) {
     $templateParameters += $AdditionalParameters
 }
+if ($context.Environment.StorageEndpointSuffix) {
```

Comment:
I did not see any documentation that discussed what parameters were available to templates, so I am assuming that it is safe to extend.   We may want to consider enumerating the members of `$context.Environment` and writing each out for the template to opt-into.  Though there are quite a few that don't seem useful in the context of an ARM template, so I left this scoped to just the core scenario.

Reply:
Is $context.Environment.StorageEndpointSuffix ever null?

Reply:
I'm reasonably sure that many of the Environment members aren't available within the Azure Stack environment.  Daniel mentioned in passing that we may be looking to do some testing on Stack in the future.

Reply:
@danieljurek can correct me but parameters that are passed can vary so we might want to only add it if this parameter isn't already passed in.

Reply:
I did not see it being passed, nor was the default in my test template overridden until I added this here.   I'll defer to @danieljurek  as the authority here, though, as I may be overlooking something in the wider environment.

Reply:
Properties in PowerShell don't throw, so it doesn't matter if `$context` or `$context.Environment` is null. The result will be `$null` in any case, so this is totally safe.

Reply:
For context, my intent in placing this within an `if` is to ensure that we don't add the parameter with an empty value if it isn't populated for the current cloud environment, so that any default for the parameter within the template is applied.

Reply:
We've been passing `storageEndpointSuffix` as an ARM template parameter in smoke tests. Setting it based on the context is perfectly fine for the use cases I can think of and is preferable because we don't have to fill our YAMLs up with well-known configuration values. 

There is a problem here that if `$templateParameters` already has an entry for `storageEndpointSuffix` from `$AdditionalParameters` then the collision will result in an error. As written this would break the .NET smoke tests because they specify `storageEndpointSuffix` in https://github.com/Azure/azure-sdk-for-net/blob/master/common/SmokeTests/smoke-tests.yml#L54

To fix it might make sense to check that there isn't already a value at that key.

Reply:
Nice catch;  I hadn't considered that scenario.   I've added a `ContainsKey` guard to ensure that we prioritize the `$AdditionalParameters` and do not try to inject the environment version if it has been set.